### DISCUSSION TITLE
Handlebars or Mustache example needed

### DISCUSSION THREAD

**Discussion by patlecat** [taocpp/PEGTL#276] [Ideas]
I would appreciate if someone could make an example of a handlebars or mustache parser for parsing template code in HTML. I'm sure many would find that useful.
https://handlebarsjs.com/

**Comment by d-frey** [comment]
You asked that question before, #105.

**Comment by patlecat** [comment]
Yeah, thought it might inspire someone if it comes up again.

**Reply by d-frey** [reply]
OK, let's see. But please don't make this a regular occurance.

**Reply by patlecat** [reply]
yeah i understand. which other parser generator would be good for that task (using C++) especially for a beginner?

**Reply by d-frey** [reply]
Given that you want/need more than just a grammar (as you said in #105), don't look for parser generators. You want a full-featured mustache/handlebars library.

**Reply by patlecat** [reply]
I know where I am here, and a grammar for this would be great to have also for this community. No one needs just a grammar in their application though, that is self-evident in the end I'll need a fully functional parser out of it.

**Reply by ColinH** [reply]
To be very clear, a Mustache/Handlebars grammar could be in-scope for the PEGTL and live in contrib and, time permitting, we would give some support if you, or anybody else, were to implement one.

A full Mustache/Handlebars processor is _wildly_ out-of-scope and could, at best and when simplified*, be added to our JSON library where it would benefit from all the infrastructure for the JSON data model.

   * Simplified because Handlebars supports "helpers" written in JavaScript so full support needs an embedded JavaScript interpreter...
